src/main/java/frc/utils/controllers/keyboard/KeyboardController.java

```diff
@@ -0,0 +1,142 @@
+        this.X = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/x", false)::get);
+        this.C = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/c", false)::get);
+        this.V = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/v", false)::get);
+        this.B = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/b", false)::get);
+        this.N = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/n", false)::get);
+        this.M = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/m", false)::get);
+
+        this.SEMICOLON = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/;", false)::get);
+        this.APOSTROPHE = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/'", false)::get);
+        this.LEFT_SHIFT = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/shift", false)::get);
+        this.COMMA = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/,", false)::get);
+        this.PERIOD = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/.", false)::get);
+        this.SLASH = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/slash", false)::get);
+        this.RIGHT_SHIFT = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/right shift", false)::get);
+        this.LEFT_CONTROL = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/ctrl", false)::get);
+        this.LEFT_ALT = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/alt", false)::get);
+        this.RIGHT_CONTROL = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/right ctrl", false)::get);
+
+        this.LEFT_ARROW = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/left", false)::get);
+        this.RIGHT_ARROW = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/right", false)::get);
+        this.UP_ARROW = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/up", false)::get);
+        this.DOWN_ARROW = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/down", false)::get);
+
+        this.NUMPAD_0 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad0", false)::get);
+        this.NUMPAD_1 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad1", false)::get);
+        this.NUMPAD_2 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad2", false)::get);
+        this.NUMPAD_3 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad3", false)::get);
+        this.NUMPAD_4 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad4", false)::get);
+        this.NUMPAD_5 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad5", false)::get);
+        this.NUMPAD_6 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad6", false)::get);
+        this.NUMPAD_7 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad7", false)::get);
+        this.NUMPAD_8 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad8", false)::get);
+        this.NUMPAD_9 = new Trigger(new LoggedTableBoolean("Keyboard", "Keys/numpad9", false)::get);
```

Comment:
make keyboard and keys a const, and think if you can make the ither usages of keys and keyboard with same const